### PR TITLE
feat(payment-service): add paypal payment gateway

### PR DESCRIPTION
**Pull request by shivamSF**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the style guide
- [x] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shubhamp-sf — APPROVED** [review]

File: sandbox/payment-example/backend/package.json [hunk 1]
```diff
@@ -53,16 +53,17 @@
     "!*/__tests__"
   ],
   "dependencies": {
-    "@loopback/boot": "^5.0.2",
-    "@loopback/context": "^5.0.2",
-    "@loopback/core": "^4.0.2",
-    "@loopback/repository": "^5.0.2",
-    "@loopback/rest": "^12.0.2",
-    "@loopback/rest-explorer": "^5.0.2",
-    "@loopback/service-proxy": "^5.0.2",
```
Comment (shubhamp-sf) [inline sandbox/payment-example/backend/package.json:62]: @shivamSF Was keeping `^` in versions causing any issues?

Comment (shivamSF) [reply sandbox/payment-example/backend/package.json:62]: yes, but we can add them back as it was while creating package-lock, now that it has been created should not be a problem. will update the PR

Comment (shubhamp-sf) [reply sandbox/payment-example/backend/package.json:62]: Thanks.

File: services/payment-service/src/providers/paypal/paypal-provider.service.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+      throw new Error('Paypal config missing');
+    }
+    if (process.env.PAYPAL_ENVIRONMENT === 'sandbox') {
+      this.environment = new paypal.core.SandboxEnvironment(
+        this.config?.clientId,
+        this.config?.clientSecret,
+      );
+    }
+    if (process.env.PAYPAL_ENVIRONMENT === 'live') {
+      this.environment = new paypal.core.LiveEnvironment(
```
Comment (yeshamavani) [inline services/payment-service/src/providers/paypal/paypal-provider.service.ts:49]: why is this needed?

Comment (shivamSF) [reply services/payment-service/src/providers/paypal/paypal-provider.service.ts:49]: paypal has two configurations live and sandbox, sandbox creds dont work in live and vice versa
